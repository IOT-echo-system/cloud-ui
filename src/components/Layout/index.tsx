import type {PropsWithChildren} from 'react'
import React from 'react'

const Layout: React.FC<PropsWithChildren> = ({children}) => {
  return <div>{children}</div>
}

export default Layout
