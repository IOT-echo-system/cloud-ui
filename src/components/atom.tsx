import {styled, Button as MUIButton} from '@mui/material'

export const Button = styled(MUIButton)({textTransform: 'none'})
