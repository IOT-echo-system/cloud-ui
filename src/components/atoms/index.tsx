export * from './StyledComponents'
export * from './FormInput'
export * from './ToastWrapper'
export * from './Loader'
export * from './CollapsibleAccordions'
