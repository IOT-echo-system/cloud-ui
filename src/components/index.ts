export {default as Layout} from './Layout'
export * from './Layout'

export {default as SiteWrapper} from './SiteWrapper'
export * from './SiteWrapper'

export * from './atom'
