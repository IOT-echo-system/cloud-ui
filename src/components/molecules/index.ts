export * from './Header/Header'
export * from './Footer'
export * from './Form'
export * from './ValidatedProfile'
export * from './FetchDetails'
export * from './CustomModal'
export * from './BoardSecretKey/BoardSecretKey'
export * from './ConfirmationModals'
export * from './MqttSubscriber'
