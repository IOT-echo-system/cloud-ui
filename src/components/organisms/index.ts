export * from './Layout'
export * from './BoardDetails'
export * from './RoutineDetails'
export * from './ModalForms/ModalForms'
export * from './Widget'
