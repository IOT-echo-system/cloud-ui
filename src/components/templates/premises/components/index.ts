export * from './BoardsView'
export * from './FeedsView'
export * from './PremisesOverview'
export * from './RoutinesView'
export * from './UsersView'
export * from './Widgets'
export * from './ZonesView'
