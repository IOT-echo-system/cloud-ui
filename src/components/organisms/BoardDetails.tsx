import {Fab, Stack} from '@mui/material'
import type {Board} from '../../typing/board'
import React from 'react'
import {PolicyUtils} from '../../utils/policyUtils'
import {Key} from '@mui/icons-material'
import {PolicyAllowed, WidgetsContainer} from '../atoms'
import {BoardSecretKey, CustomModal} from '../molecules'

type BoardDetailsPropsType = {board: Board}
export const BoardDetails: React.FC<BoardDetailsPropsType> = ({board}) => {
  return (
    <Stack>
      <Stack direction={{xs: 'column', sm: 'row'}} alignItems={'start'} justifyContent={'space-between'} spacing={2}>
        {/*<EditableTitleWithId*/}
        {/*  policyId={PolicyUtils.BOARD_UPDATE}*/}
        {/*  getFormDetails={EditBoardName}*/}
        {/*  board={board}*/}
        {/*  title={board.name}*/}
        {/*  id={board.boardId}*/}
        {/*  idLabel={'Board Id'}*/}
        {/*/>*/}
        <Stack direction={{xs: 'row-reverse', sm: 'row'}} spacing={2} alignItems={'center'}>
          <PolicyAllowed policyId={PolicyUtils.DEVICE_UPDATE}>
            <CustomModal
              ClickableComponent={
                <Fab color={'primary'} size={'small'}>
                  <Key />
                </Fab>
              }
            >
              <BoardSecretKey board={board} />
            </CustomModal>
          </PolicyAllowed>
          {/*<Chip*/}
          {/*  label={board.status}*/}
          {/*  color={board.status === 'HEALTHY' ? 'success' : 'error'}*/}
          {/*  sx={{padding: {xs: '0', md: '8px 32px'}}}*/}
          {/*/>*/}
        </Stack>
      </Stack>
      <WidgetsContainer></WidgetsContainer>
    </Stack>
  )
}
