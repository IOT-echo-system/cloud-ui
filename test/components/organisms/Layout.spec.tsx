import {render} from '@testing-library/react'
import React from 'react'
import {Layout} from '../../../src/components/organisms'

describe('Layout component test', () => {
  it('should match with the snapshot', () => {
    const {container} = render(<Layout />)

    expect(container).toMatchSnapshot()
  })
})
