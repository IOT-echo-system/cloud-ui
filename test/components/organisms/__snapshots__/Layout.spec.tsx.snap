// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component test should match with the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1gfrmn8-MuiStack-root"
  />
</div>
`;
